http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/db/TestComponent.scala

```diff
@@ -2,23 +2,60 @@ package org.broadinstitute.dsde.workbench.leonardo.db
 
 import java.time.Instant
 
+import cats.effect.concurrent.Semaphore
+import cats.effect.{IO, Resource}
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.leonardo.config.LiquibaseConfig
 import org.broadinstitute.dsde.workbench.leonardo.model.Cluster
 import org.broadinstitute.dsde.workbench.leonardo.model.google.ClusterName
 import org.broadinstitute.dsde.workbench.leonardo.{CommonTestData, GcsPathUtils, LeonardoTestSuite, RuntimeConfig}
 import org.broadinstitute.dsde.workbench.model.google.{GoogleProject, ServiceAccountKeyId}
 import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.time.{Seconds, Span}
+import org.scalatest.{BeforeAndAfterAll, TestSuite}
+import slick.basic.DatabaseConfig
 import slick.dbio.DBIO
+import slick.jdbc.JdbcProfile
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration._
 
-trait TestComponent extends LeonardoTestSuite with ScalaFutures with GcsPathUtils {
+trait TestComponent extends LeonardoTestSuite with ScalaFutures with GcsPathUtils with BeforeAndAfterAll {
+  this: TestSuite =>
   implicit override val patienceConfig = PatienceConfig(timeout = scaled(Span(10, Seconds)))
 
   val defaultServiceAccountKeyId = ServiceAccountKeyId("123")
 
-  implicit val dbRef = DbSingleton.dbRef
+  val initWithLiquibaseProp = "initLiquibase"
+  val liquiBaseConfig =
+    LiquibaseConfig("org/broadinstitute/dsde/workbench/leonardo/liquibase/changelog.xml", true)
+
+  // Not using beforeAll because the dbRef is needed before beforeAll is called
+  implicit protected lazy val dbRef: DbRef[IO] = initDbRef.unsafeRunSync()
+
+  override def afterAll(): Unit = {
+    dbRef.close()
+    super.afterAll()
+  }
+
+  // This is a bit duplicative of DbReference.init but that method returns a Resource[F, DbRef[F]]
+  // which doesn't play nicely with ScalaTest BeforeAndAfterAll. This version returns an F[DbRef[F]].
+  private def initDbRef: IO[DbRef[IO]] =
+    for {
+      concurrentPermits <- Semaphore[IO](100)
```

Comment:
Should `100` come from a config?

Reply:
Ah yeah good catch, this is already in reference.conf